specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json

```diff
@@ -892,7 +892,39 @@
         "path"
       ]
     },
-    "SqlServerTableDataset": {
+    "SapEccResourceDataset": {
+      "x-ms-discriminator-value": "SapEccResource",
+      "description": "The path of the SAP ECC OData entity.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Dataset"
+        }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "SAP ECC OData resource dataset properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SapEccResourceDatasetTypeProperties"
+        }
+      },
+      "required": [
+        "typeProperties"
+      ]
+    },
+    "SapEccResourceDatasetTypeProperties": {
+      "description": "Sap ECC OData resource dataset properties.",
+      "properties": {
+        "path": {
+          "type": "object",

```

Comment:
This should be type "string". In your description, it seems like this can either be a string or a delegate that returns string - but the latter is not serializable.

Reply:
Thanks for your review. This is align with the other Dataset, such as the SalesforceObjectDatasetTypeProperties or SAPCloudForCustomer. Could you please advise why these can exist while this one cannot? Thanks.

Reply:
Then the other Dataset is incorrect too. Two reasons:
1. In the JSON Schema type "object" means a key value pair formatted actual object. A string doesn't qualify. We had some confusions around this before, which might caused the other dataset error to be overlooked.
2. As you suggested in the comments, this can be either a string or a delegate. In the latter case a delegate is not serializable so you cannot send it on the wire anyways. Why not just invoke it and send the string result?

Reply:
About the first bullet point, see more at https://github.com/Azure/autorest/issues/2827

Reply:
thanks Jinghao. I made the changes as advised.